### ISSUE TITLE
View buttons crashing

### ISSUE THREAD

**Issue by xJosHubx** [freeCodeCamp/mobile#1179]
**Describe the bug** 
When I am cycling between the html viewer and the page viewer (the i and the eye) so that I can look at my project while I'm coding, I get stuck on the html viewer. The icons light up white when selected normally. In this case the "eye" icon will not light up white and will not display the page that I'm working on, making it impossible to see my changes without closing and reopening the app. After closing and reopening the app I can then select the "eye" icon and view my page until I make a change to the html and then it all happens over again.  

**To Reproduce**
1. Open the app
2. Click on 'eye' icon
3. Click on the html
4. Make any change  
5. Now try and click on the 'eye' icon

**Expected behavior**
It should not light up the "eye" icon. You should be unable to see the page view. Clicking back and forth between icons for html and page view should crash page viewer so that it becomes unresponsive and you are unable to open page viewer. 

**Screenshots**

**Smartphone (please complete the following information):**

**Additional context**
I do the majority of my coding on iPad M2 pro that I bought in Nov, 2023. And I've been doing all my FCC app on my iPad as well.

**Comment by Sembauke** [comment]
Hey @xJosHubx,

Thank you for reporting this. Could you share a video with us? The instructions are somewhat hard to follow.

**Comment by xJosHubx** [comment]
I'm not sure if someone worked on this issue but I'm not able to reproduce it anymore. Will keep you posted. Thank you so much for your response!

**Comment by xJosHubx** [comment]
Hey @sembauke here is the video of this issue https://youtu.be/ID_cKUDgnco?si=ZJzcrdCEziGcW0N5. Thank you so much for working on this! I really want to have a good UX while using the app. 😊

**Comment by Sembauke** [comment]
Thank you for the update and the video. It seems that there are many issues on large devices.

**Comment by xJosHubx** [comment]
Thank you, Sem. I'll do my best.
